nixos/modules/services/misc/omnom.nix

```diff
@@ -0,0 +1,182 @@
+                default = "sqlite";
+                description = "Database type.";
+              };
+            };
+            server = {
+              address = lib.mkOption {
+                type = lib.types.str;
+                default = "127.0.0.1:${toString cfg.port}";
+                description = "Server address.";
+                defaultText = lib.literalExpression ''
+                  "127.0.0.1:''${config.services.omnom.port}"
+                '';
+              };
+              secure_cookie = lib.mkOption {
+                type = lib.types.bool;
+                default = true;
+                description = "Limit cookies to secure channel.";
+              };
+            };
+            storage = {
+              root = lib.mkOption {
+                type = lib.types.str;
+                default = "${cfg.dataDir}/static/data";
+                defaultText = lib.literalExpression ''
+                  "''${config.services.omnom.dataDir}/static/data"
+                '';
+                description = "Where the snapshots are saved.";
+              };
+            };
+          };
+        };
+        default = { };
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    environment.etc."omnom/config.yml".source = settingsFormat.generate "omnom-config.yml" cfg.settings;
+
+    services.omnom.settings = {
+      app.debug = lib.mkDefault cfg.debug;
+      storage.type = lib.mkDefault "fs";
+    };
+
+    systemd.services.omnom = {
+      path = [ pkgs.rsync ];
+
+      serviceConfig = {
+        DynamicUser = isDynamicUser;
+        User = cfg.user;
+        Group = cfg.group;
+        ExecStart = ''
+          ${lib.getExe cfg.package} listen \
+            --config /etc/omnom/config.yml \
+            ${lib.optionalString cfg.debug "-d"}
+        '';
+        StateDirectory = lib.mkIf (cfg.dataDir == "/var/lib/omnom") "omnom";
+        StateDirectoryMode = lib.mkIf (cfg.dataDir == "/var/lib/omnom") "0750";
+        WorkingDirectory = cfg.dataDir;
+        Restart = "on-failure";
+        RestartSec = "10s";
+      };
+      after = [
+        "network.target"
+        "systemd-tmpfiles-setup.service"
+      ];
+      wantedBy = [ "multi-user.target" ];
+
+      # These need to be present in the working directory
+      preStart = ''
+        for dir in templates; do
+          rsync -a --chmod=u=rwX,go=rX "${cfg.package}/share/$dir" "${cfg.dataDir}/"
+        done
+      '';
+    };
+
+    # If the service user is not dynamic, a normal group must exist
+    users.groups = lib.mkIf ((!isDynamicUser) && (cfg.group == "omnom")) { omnom = { }; };
+
+    systemd.tmpfiles.settings."10-omnom" = lib.mkIf (cfg.dataDir != "/var/lib/omnom") (
+      {
+        "${cfg.settings.storage.root}/resources"."d" = {
+          inherit (cfg) user group;
+          mode = "0750";
+        };
+        "${cfg.settings.storage.root}/snapshots"."d" = {
+          inherit (cfg) user group;
+          mode = "0750";
+        };
```

Comment:
As stated above I don't think we actually need to create these directories with that specific mode.